### PR TITLE
DevEx: fix nested entity validation

### PR DESCRIPTION
**Pull request by rachaelparris**
`isValid()` was not working correctly for nested entities, but `getFormattedValidationErrors()` was. I've updated `isValid` to use `getFormattedValidationErrors` under the hood to reduce the number of ways we process validation. This also removes the need for the `customValidate` function on entities. I'll be removing those functions from all entities in another PR to keep this one focused.

### CODE REVIEW

File: shared/src/business/entities/cases/CaseExternalIncomplete.js [hunk 1]
```diff
@@ -54,9 +54,11 @@ joiValidationDecorator(
     preferredTrialCity: CaseExternal.commonRequirements.preferredTrialCity,
     procedureType: CaseExternal.commonRequirements.procedureType,
   }),
-  function () {
-    return !this.getFormattedValidationErrors();
-  },
+  //TODO - remove all customValidation functions - they are not being used anymore
```
Comment (rachaelparris) [inline shared/src/business/entities/cases/CaseExternalIncomplete.js:57]: This will be removed in another PR coming soon. I had to comment it out here for test coverage.

Comment (kkoskelin) [reply shared/src/business/entities/cases/CaseExternalIncomplete.js:57]: There were some situations in which I would have really appreciated the ability to write custom validation functions.  See the `OrderSearch.js` entity, one that is not persisted but has some funny kind of requirements that are difficult to express with `joi` (sounds like the code needs a therapist...).  Maybe we don't have to create/preserve functionality right here and in this way, but something to think about for the future.

File: shared/src/business/useCases/saveCaseDetailInternalEditInteractor.test.js [hunk 2]
```diff
@@ -159,25 +155,21 @@ describe('updateCase', () => {
...
           contactPrimary: null,
           contactSecondary: {},
         },
-      });
-    } catch (err) {
-      error = err;
-    }
-    expect(error).toBeNull();
+      }),
+    ).rejects.toThrow();
```
Comment (kkoskelin) [inline shared/src/business/useCases/saveCaseDetailInternalEditInteractor.test.js]: Should this throw anything in particular?

**Review by kkoskelin — CHANGES_REQUESTED** [review]
This is really important, and excellent. Thank you!
Just one note where I'd like to be more specific about what error is being thrown.

File: shared/src/business/entities/cases/CaseExternalIncomplete.js (hunk 1 above)
Comment (rachaelparris) [reply shared/src/business/entities/cases/CaseExternalIncomplete.js:57]: I think there are some more things we can do, but the way this is written now it's just not working correctly and not super clear either. We can expand on this some more for sure to meet our needs.

**Review by kkoskelin — APPROVED** [review]
